### ISSUE TITLE
jquery datepicker inside a lightbox (modal)

### ISSUE THREAD

**Issue by zomby22** [wet-boew/wet-boew#7612]
Is it possible to include a date picker field inside a modal box. Bootstrap WET 4.

``
<section id="name" class="mfp-hide modal-dialog modal-content overlay-def">
	<header class="modal-header">
		<h2 class="modal-title">title</header>
	<div class="modal-body">
		form with date picker field here.
       <input type="date" value=""/>
	</div>
</section>``

**Comment by RobJohnston** [comment]
VERY closely related to #7611, where the problem was only identified in IE 11.

**Comment by hsrudnicki** [comment]
Is this still an issue? Please update the status of this again, whether this can be closed or otherwise, thank you

**Comment by zomby22** [comment]
this somewhat answer my question. If there is a way to force the "modal-dialog" closing by pressing the X only instead of it close if you click anyone on the popup page, this will solve the date picker to stay open along with the modal box. If anyone has an answer, let me know, otherwise it can be closed.

**Comment by kpduncan** [comment]
I have a solution! Also a live working demo! Might of lost my hair with figuring this one out. The issue is on IE it is using WET date picker. The WET date picker is an overlay modified and populated to the page with WET enchantment library. The key to getting this to work was moving populated code to the overlay and changing some css in a timely mater. Here is the html:
```html
<a id="button1" class="btn btn-primary wb-lbx lbx-modal" href="#popup">Center Overlay/Modal</a> 
<section id="popup" class="modal-dialog modal-content overlay-def hidden">
  <div id="popup-bdy" tabindex="0">
    <header class="modal-header">
      <h2 class="modal-title">Center Overlay/Modal</h2>
    </header>
    <div class="modal-body">
      <form action="#">
        <div class="form-group" id="date-div">
          <label for="date">Date Example:</label>
          <input id="date" class="form-control" type="date" name="date" placeholder="Enter date..."/>
        </div>
      </form>
      <button class="btn btn-md btn-primary popup-modal-dismiss" tabindex="0"
      role="button">Close</button>
    </div>
  </div>
</section>
```
The JavaScript their is two file (modal-script.js)[https://www.cybersubtype.ca/wet-theme/js/modal-script.js] and (ie-modal-datepicker.js)[https://www.cybersubtype.ca/wet-theme/js/ie-modal-datepicker.js] these files are required. Also the (live demo)[https://www.cybersubtype.ca/apps/wet-demos/] use the first button. Thanks!

**Comment by zomby22** [comment]
thanks, working! in my example, I use a database table that loops multiples rows of data. Each row has the edit link that opens the modal popup overlay.

**Comment by kpduncan** [comment]
@hsrudnicki You may be able to close 7611 and 7545 I think their pretty much the same.

**Comment by LaurentGoderre** [comment]
That's not an ideal solution. We should fix the calendar to work with overlays

**Comment by sival13** [comment]
@ghost The site where you posted your solution is not working.
Can you please post the fix ?

Thanks

**Comment by kodecount** [comment]
@LaurentGoderre I can offer a start on a actual fix? It works except need to create a function to determine the height of the overlay body. Other then that it completely functional.

**Comment by kodecount** [comment]
@LaurentGoderre @sival13 Please view [PR 7770](https://github.com/wet-boew/wet-boew/pull/7770)

**Comment by sival13** [comment]
Here is the issues I have found and how I resolved them.

1) datepicker.css: z-index is too low 
line 24:  z-index: 1050;
2) Date is not returned back into the input field
datepicker.js : line 256 - 266 - This prevents the actual date to be returned in the input field.
Does not seem to have any effect when there is no lightbox
$document.on( "focusout focusin", "#" + containerName + " .wb-clndr",  function( event ) {

	// Hide the calendar when the focus leaves the calendar
	switch ( event.type ) {
		case "focusout":
			focusOutTimer = setTimeout( hide, 10 );
			break;
		case "focusin":
			clearTimeout( focusOutTimer );
	}
} );

3) interactions between lightbox and the date picker
I could not get to the root cause.
But when you click a date on the calendar, the light box closes.
 adding the class  popup-modal-dismiss to the X close button. Resolves this issues
 
In a lighbox link, add in the data-wb-lbx attribute as per documentation :

"closeMarkup" : "<button title=\"%title%\" type=\"button\" class=\"mfp-close popup-modal-dismiss\">×</button>"}

This modifies the close button mark up to include the popup-modal-dismiss class, and resolve what I believe to be conflicts between events.


 4) following this workflow also is problematic :
- opening a date picker on a page
- opening a lightbox (this closes the date picker)
- opening a date picker in the lighbox -> does not show the date picker as its property aria-hidden has not be set to true on closing. You therefore need to click on it twice to have it appear.

Hope this helps someone else.
